tds/src/main/java/thredds/server/wfs/WFSLogger.java

```diff
@@ -0,0 +1,42 @@
+package thredds.server.wfs;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A logger specifically used for logging events related to the WFS servlet.
+ * 
+ * @author wchen@usgs.gov
+ *
+ */
+public class WFSLogger {
+
+	static private final Logger wfsLog = LoggerFactory.getLogger(WFSController.class);

```

Comment:
Good use of SLF4J!

Reply:
Ah, I missed this on the first pass...
SLF4J wants the current class to be passed to `LoggerFactory.getLogger`. SLF4J has [a warning](https://www.slf4j.org/codes.html#loggerNameMismatch) if the name of the Logger doesn't match the name of the class from which it is called.

Reply:
Actually, I'm wondering now: is this class is needed? Would it suffice to instantiate an SLF4J `Logger` in each class that needs to do logging? `Logger`s are pretty cheap objects, so there's not much performance overhead in creating a new one per class. Adopting a one-logger-per-class practice permits frameworks like log4j to adjust the verbosity of logging on a per-class or per-package basis.

Remember when the dev VM's disk was eaten by a rogue logger on Jenkins? If the project in question used SLF4J and one-logger-per-class we could have resorted to shutting off logging for the one chatty class.

Reply:
The idea of using the WFS Logger specifically was to allow for a single SLF4J log for WFS related elements which is what is being done with DAP currently (see DapLog.class). Having a single "wfs" logger would be a good way to consolidate related messages (relating to wfs). Could certainly consider changing to a single "wfs" log or doing away with SLF4J in general for the time being considering your feedback below.

Reply:
For sure, that is a benefit of having more granular control of the logging, but also brings up another point, whether or not logging is needed as of now. Given a small VM size, it could possibly cause denial of service by filling up the storage with SLF4J logs through malicious requests! Reverting to a single RuntimeException(...) as you posed earlier.

Reply:
Cool! Makes Sense!